tstate/tstate_view.go

```diff
@@ -213,8 +213,8 @@ func (ts *TStateView) isUnchanged(ctx context.Context, key string, nval []byte,
 // Insert allocates and writes (or just writes) a new key to [tstate]. If this
 // action returns the value of [key] to the parent view, it reverts any pending changes.
 func (ts *TStateView) Insert(ctx context.Context, key []byte, value []byte) error {
-	// Inserting requires a Write Permissions, so we pass state.Write
-	if !ts.checkScope(ctx, key, state.Write) {
+	// To insert, the key needs to have Allocate or Write permissions, or both
+	if !ts.checkScope(ctx, key, state.Allocate) && !ts.checkScope(ctx, key, state.Write) {
```

Comment:
Is there a way to just do Allocate | Write for this check rather than calling checkScope twice?

Reply:
Perhaps the `perm` parameter of `checkScope` could be variadic? I introduced a change in `checkScope` that uses this, but I'm not sure if that is the cleanest way. Another way is to maybe support a helper function called `Contains` in `state`